docs/puzzles/singletons.md

```diff
@@ -0,0 +1,437 @@
+---
+id: singletons
+title: Singletons
+---
+
+One of the most important puzzles in the Chia ecosystem is the **singleton**.  This is a puzzle that assures anyone who looks at can see that it has a unique ID that no other coin has.  Parties can decide to accept messages or commitments from that unique ID with the assurance that the party who controls the singleton is not double dipping or impersonating someone else.
+
+This puzzle is an outer puzzle and is used to wrap pooling puzzles, NFTs, and decentralized identities.  Any inner puzzle can be wrapped with this puzzle if it has necessity for uniqueness.
+
+## Design choices
+
+A few design choices were made in the creation of this puzzle so let's go over them now:
+
+* **Singletons are always odd.**  In order to assure that a singleton does not duplicate itself, it needs some way to verify that it's children do not consist of more than one new singleton.  It does this by verifying that only one of its children is odd.  It is either a new singleton, or it is the **melt value** and will be ignored (more on that later).  Odd was chosen over even because you may want to have singletons create other, non-singleton coins at times (like 0 amount messages for the DID) and having the singleton be odd just makes this easier.  Coins can be multiples of 10 so you can send a full XCH rather than an XCH and a mojo.  No matter how many even amounts you subtract from an odd amount, the end result will always be odd.
```

Comment:
`s/it's/its/`

Reply:
Odd is better because O = O + E, O = O + E + E, O = O + E + E + E + ... + E, ie. an odd plus as many evens as you want. However, E = E, E = E + O + O, E = E + O + O + O + O, ie. if we did it the other way, we'd need an EVEN number of odds, which is a lot more limiting.